### PR TITLE
Feature/devsu 1416 pathway legend updates

### PR DESCRIPTION
**Pull request by Anna-log7**
This PR is for changing the pathway legend page to react, and adding support for custom pathway images if the option is specified in the report upload.
Available at `/pathway-analysis`

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
     "no-multiple-empty-lines": "warn",
     "no-this-before-super": "warn",
     "no-unreachable": "warn",
-    "no-unused-vars": "warn",
+    "no-unused-vars": "off",
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply .eslintrc.json:64]: Yes. I looked it up and the `@typescript-eslint` plugin covers these and it's recommended to disable the base rules

**Comment by kttkjl** [comment]
Noticed the checkmark on click for "UPLOAD PATHWAY LEGEND", it does the async timeout for "loading" but this is just opening up a window for users to choose a file for upload

**Review by kttkjl — CHANGES_REQUESTED** [review]
Noticed the checkmark on click for "UPLOAD PATHWAY LEGEND", it does the async timeout for "loading" but this is just opening up a window for users to choose a file for upload, and nothing about the image upload in-progress 

edit: you can circumvent the problem of no spinning loading circles by having a cleanup in AsyncButton's useEffect block

```javascriptReact  
useEffect(() => {
    let timeoutReturn;
    if (!isLoading && loadingStarted) {
      timeoutReturn = window.setTimeout(() => setLoadingStarted(false), 3000);
    }
    return () => clearTimeout(timeoutReturn);
  }, [isLoading, loadingStarted]);
```

otherwise PR looks good

**Comment by Anna-log7** [comment]
> 
> 
> Noticed the checkmark on click for "UPLOAD PATHWAY LEGEND", it does the async timeout for "loading" but this is just opening up a window for users to choose a file for upload, and nothing about the image upload in-progress
> 
> edit: you can circumvent the problem of no spinning loading circles by having a cleanup in AsyncButton's useEffect block
> 
> ```
> useEffect(() => {
>     let timeoutReturn;
>     if (!isLoading && loadingStarted) {
>       timeoutReturn = window.setTimeout(() => setLoadingStarted(false), 3000);
>     }
>     return () => clearTimeout(timeoutReturn);
>   }, [isLoading, loadingStarted]);
> ```
> 
> otherwise PR looks good

On second look, I don't think the `AsyncButton` should be used here. The button depends on the `onClick` event, but that will fire before the file is selected. I'll use a regular `Button` instead in this case

**Review by kttkjl — DISMISSED** [review]

**Review by kttkjl — DISMISSED** [review]

**Comment by kttkjl** [comment]
Replacing the async button fixes the clicking problem and the problem you described, but the user won't have an indicator whether or not their custom legend is being uploaded, the async progress should be shown somewhere else

**Review by kttkjl — APPROVED** [review]

**Pull request merged**